client/src/components/NewBoardGameForm.js

```diff
@@ -17,6 +17,11 @@ const NewBoardGameForm = props => {
 	const [errors, setErrors] = useState([])
 	const [shouldRedirect, setShouldRedirect] = useState(false)
 
+	const [showButton, setShowButton] = useState(false)
```

Comment:
toggling show button does not need state - the logic you have in the useEffect and exist in this component and will be evaluated on each render